### PR TITLE
Beautify ISAAC's mixing arithmetic

### PR DESCRIPTION
**Pull request by mmcco**
Call me verbose, but the existing logic was hideously formatted. The
result still fits in a small terminal. No functional change.

### CODE REVIEW

File: src/librand/isaac.rs [hunk 1]
```diff
@@ -77,14 +77,37 @@ impl IsaacRng {
...
-                b=b^(c>>2);  e=e+b; c=c+d;
-                c=c^(d<<8);  f=f+c; d=d+e;
-                d=d^(e>>16); g=g+d; e=e+f;
-                e=e^(f<<10); h=h+e; f=f+g;
-                f=f^(g>>4);  a=a+f; g=g+h;
-                g=g^(h<<8);  b=b+g; h=h+a;
-                h=h^(a>>9);  c=c+h; a=a+b;
+                a = a ^ (b << 11);
+                d = d + a;
+                b = b + c;
```
Comment (eddyb) [inline src/librand/isaac.rs]: Why aren't all of these `a ^= b << 11;`, `d += a;`, `b += c;` etc.?

Comment (mmcco) [inline src/librand/isaac.rs]: I initially tried that, but apparently arithmetic assignment operators aren't defined for Wrapping types like these. I imagine they will be eventually - I can't imagine why that wouldn't be simple.

Comment (eddyb) [inline src/librand/isaac.rs]: Oh, I didn't realize those were `Wrapping`. I guess that's a bug, because we have the traits now.

**Pull request merged**